libusb/os/darwin_usb.c

```diff
@@ -648,7 +648,7 @@ static void darwin_devices_detached (void *ptr, io_iterator_t rem_devices) {
         if (old_device->in_reenumerate) {
           /* device is re-enumerating. do not dereference the device at this time. libusb_reset_device()
            * will deref if needed. */
-          usbi_dbg (NULL, "detected device detached due to re-enumeration. sessionID: 0x%" PRIx64 ", locationID: 0x%" PRIx64,
+          usbi_dbg (NULL, "detected device detached due to re-enumeration. sessionID: 0x%" PRIx64 ", locationID: 0x%x",
```

Comment:
Is it actually correct to use %x with uint32_t ? Or is PRIx32 needed (or cast to unsigned long)?

Reply:
Hm, that's a good point.
To be strictly correct with `x` it has to be `unsigned int` (not `long`).
For `uint32_t` `PRIx32` is needed indeed.

And I believe this is not the only place in this file where similar issue exists: in the `darwin_get_cached_device` the `UInt32 locationID` is printed as `PRIx64` which doesn't look right too.

Please doeble-check and update in both places, since we're touching the `locationID` here.

Reply:
Yep, I meant unsigned long with %lx :) Is also guaranteed to work.

Reply:
> in the `darwin_get_cached_device` the `UInt32 locationID` is printed as `PRIx64`

You mean printed as %x , right?

Reply:
I'll fix up both places.

Reply:
>Is it actually correct to use %x with uint32_t ? 

It is correct. Because %x is for (unsigned) int generally, and on darwin that's guaranteed to be 32 bit. That will never change because it would totally break ABI.

Reply:
You don't mind me changing it to PRIx32 for symmetry reasons and potential consistency with non-darwin files?

Reply:
Thanks, I had already changed it there and in darwin_get_cached_device, and tested on macOS (normal build).

Reply:
> on darwin that's guaranteed to be 32 bit

Our of curiocity, can you point to the documentation where that is guaranteed?

Reply:
https://developer.apple.com/library/archive/documentation/Darwin/Conceptual/64bitPorting/transition/transition.html

https://developer.apple.com/documentation/xcode/application-binary-interfaces